src/Build/Evaluation/Conditionals/Scanner.cs

```diff
@@ -333,24 +334,40 @@ private static bool ScanForPropertyExpressionEnd(string expression, int index, o
                         if (character == '(')
                         {
                             nestLevel++;
-                            whitespaceCheck = true;
                         }
                         else if (character == ')')
                         {
                             nestLevel--;
-                            whitespaceCheck = false;
                         }
-                        else if (whitespaceCheck && char.IsWhiteSpace(character) && ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave16_10))
+                        else if (char.IsWhiteSpace(character))
                         {
+                            whitespaceFound = true;
                             indexResult = index;
-                            return false;
+                        }
+                        else if (!XmlUtilities.IsValidSubsequentElementNameCharacter(character))
+                        {
+                            nonIdentifierCharacterFound = true;
+                        }
+
+                        if (character == '$')
```

Comment:
I think you need to add this lookahead to handle the test case with a literal `$(` in it.

```suggestion
                        if (character == '$' && index < expression.Length - 1 && pchar[index + 1] == '(')
```